pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -356,6 +366,13 @@ func (s *Service) getOrCreateMonitor(openStackCluster *infrav1.OpenStackCluster,
 		Timeout:        5,
 	}
 	monitor, err = s.loadbalancerClient.CreateMonitor(monitorCreateOpts)
+	// Skip creating monitor if it is not supported by Octavia provider
+	if capoerrors.IsNotImplementedError(err) {
+		record.Warnf(openStackCluster, "SkippedCreateMonitor", "Health Monitor is not implemented with the current provider.")
```

Comment:
`"Health monitors are not supported with the current Octavia provider"` would sound a tiny bit better, second message can be updated in a similar fashion.